osm2gtfs/core/osm_connector.py

```diff
@@ -222,116 +242,118 @@ def get_stops(self, refresh=False):
 
         # Obtain raw data about routes from OpenStreetMap
         result = self._query_stops()
+        self.stops['regular'] = {}
+        self.stops['stations'] = {}
 
         # Build stops from ways (polygons)
         for stop in result.ways:
-            if self._is_valid_stop_candidate(stop):
-                self.stops["way/" + str(stop.id)
-                           ] = self._build_stop(stop, "way")
+            osm_type = "way"
+            stop_object = self._build_stop(stop, osm_type)
+            if stop_object:
+                self.stops['regular'][osm_type + "/" + str(
+                    stop_object.osm_id)] = stop_object
 
         # Build stops from nodes
         for stop in result.nodes:
-            if self._is_valid_stop_candidate(stop):
-                self.stops["node/" + str(stop.id)
-                           ] = self._build_stop(stop, "node")
-
-        # Build stop_areas
-        for relation in result.relations:
-            # valid stop_area candidade?
-            if 'public_transport' in relation.tags:
-                if relation.tags["public_transport"] == "stop_area":
-                    self.stops["relation/" + str(relation.id)
-                               ] = self._build_stop_area(relation)
+            osm_type = "node"
+            stop_object = self._build_stop(stop, osm_type)
+            if stop_object:
+                self.stops['regular'][osm_type + "/" + str(
+                    stop_object.osm_id)] = stop_object
+
+        # Build stations from stop_area relations
+        for stop in result.relations:
+            osm_type = "relation"
+            stop_object = self._build_station(stop, osm_type)
+            if stop_object:
+                self.stops['stations'][osm_type + "/" + str(
+                    stop.id)] = stop_object
 
         # Cache data
-        Cache.write_data('stops-' + self.selector, self.stops)
+        Cache.write_data(self.selector + '-stops', self.stops)
 
         # Maybe check for unnamed stop names
         if self.auto_stop_names:
             self._get_names_for_unnamed_stops()
 
-        # Warning about stops without stop_area
-        for ref, elem in self.stops.iteritems():
-            if type(elem) is Stop:
-                sys.stderr.write("Stop is not member of a stop_area." +
-                                 " Please fix in OpenStreetMap\n")
-                sys.stderr.write("http://osm.org/" + ref + "\n")
```

Comment:
It was removed, because Stops don't have to be part of a stop_area in OSM. I introduced this part some time ago, but I learned that most people don't do this. I saw it as a good practice but it also consists in an overhead, that I understand people don't want to maintain in OSM. So, I removed it again.